### DISCUSSION TITLE
Built-in options for NAS mutation visualization?

### DISCUSSION THREAD

**Discussion by DaanSchalk** [microsoft/nni#4526] [Q&A]
Hey everyone, I am looking for a way to visualize the chosen mutations for each trial of a NAS experiment.

When using a tuner (example: nni/examples/trials/mnist-pytorch)
It automatically shows the taken search space options for all trials in the hyperparameter tab:
<img width="830" alt="nni_vis1" src="https://user-images.githubusercontent.com/25820623/152130579-cf6f9b9f-4a7f-4177-bdf2-8a91fe3702d9.PNG">
I like this visualization because it quickly shows the sensitivity of the performance for each choice/mutation

However, when using the example given for multi-trial NAS with inline mutations (nni/examples/nas/multi-trial/mnist)
The mutations are shown in the parameter tab 
<img width="233" alt="nni_vis3" src="https://user-images.githubusercontent.com/25820623/152130644-d4698460-810e-430a-8732-84cb23aa00da.PNG">
but the visualization from the hyperparameter tab becomes less useful:
<img width="832" alt="nni_vis2" src="https://user-images.githubusercontent.com/25820623/152130671-f131bf3f-1c17-40c3-9cfc-e10a4a23ae13.PNG">

My question: Is there a built-in method for getting such a visualization for inline mutations when using the retiarii NAS experiment setup?

**Comment by ultmaster** [comment]
I believe NNI v2.6 has improved the visualization.

If upgrading NNI doesn't work for you, I believe there might be some bugs, please submit an issue.
